src/encoded/upgrade/donor.py

```diff
@@ -144,3 +144,46 @@ def fly_worm_donor_5_6_and_human_mouse_8_9(value, system):
     # http://redmine.encodedcc.org/issues/5041
     if value.get('status') in ['preliminary', 'proposed']:
         value['status'] = "in progress"
+
+@upgrade_step('fly_donor', '6', '7')
+@upgrade_step('worm_donor', '6', '7')
+@upgrade_step('mouse_donor', '9', '10')
+def model_organism_donor_9_10(value, system):
+    # https://encodedcc.atlassian.net/browse/ENCD-3415
+    value.pop('internal_tags', None)
+    value.pop('littermates', None)
+    value.pop('url', None)
+
+@upgrade_step('human_donor', '9', '10')
+def human_donor_9_10(value, system):
+    # https://encodedcc.atlassian.net/browse/ENCD-3415
+    if value.get('life_stage') == 'fetal':
+        value['life_stage'] = 'embryonic'
+    if value.get('life_stage') == 'postnatal':
+        value['life_stage'] = 'newborn'
+    if not value.get('internal_tags'):    
+        value.pop('internal_tags', None)
+    if value.get('ethnicity') in ['NA', 'Unknown', 'unknown', '']:
+        value.pop('ethnicity')
+    if value.get('ethnicity') == 'African-American':
```

Comment:
should we maybe only use one of African American and Black?

Reply:
If we have these details, why collapse?

Reply:
ok if they are subclassed.

Reply:
in any case it's "someone elses fault"